receiver/otlpreceiver/internal/logs/otlp.go

```diff
@@ -51,14 +48,7 @@ func (r *Receiver) Export(ctx context.Context, req plogotlp.ExportRequest) (plog
 	// NonPermanent errors will be converted to codes.Unavailable (equivalent to HTTP 503)
 	// Permanent errors will be converted to codes.InvalidArgument (equivalent to HTTP 400)
 	if err != nil {
-		s, ok := status.FromError(err)
-		if !ok {
-			code := codes.Unavailable
-			if consumererror.IsPermanent(err) {
-				code = codes.InvalidArgument
-			}
-			s = status.New(code, err.Error())
-		}
+		s := errors.GetStatusFromError(err)
 		return plogotlp.NewExportResponse(), s.Err()
```

Comment:
```suggestion
		return plogotlp.NewExportResponse(), errors.GetStatusFromError(err).Err()
```

Reply:
If this is the case, why not always return the error from GetStatusFromError?

Reply:
Returning the entire `Status` makes `GetStatusFromError` more flexible for future use, but since there is not ab actual use case yet, and this is an internal package, I am ok with `GetStatusFromError` returning an `error` that made from a `Status`.  

I don't feel strongly about this either way, so let me know what you all prefer.

Reply:
Since this is an internal API I prefer to simplify usage, since everywhere we call `.Err()`

Reply:
Updated